frontend/src/metabase/query_builder/components/expressions/ExpressionEditorTextfield.jsx

```diff
@@ -180,6 +180,16 @@ export default class ExpressionEditorTextfield extends React.Component {
     }
   };
 
+  // This method focuses on the editor input.
+  // Using react-ace’s `onFocus` rendered the editor code
+  // with a severe vertical misalignment.
+  // Occurrences of this bug were rare and non-deterministic wrt to browser.
+  handleEditorLoaded = () => {
+    setTimeout(() => {
+      this.input.current.editor.focus();
+    }, 10);
```

Comment:
@mazameli, this bug has been a big mystery. Seems to behave differently for different people with the same browser version, and same OS.

Can you try changing this `10` milliseconds to something like `1000`, please, and tell me if it improves?
Just so we rule out a possible race condition here.

Reply:
@mazameli, I've made a change so maybe just trying again might do it.

Reply:
Cool, I'll take a look.